File: client/src/lib/money.ts
export function formatCents(cents: number, currency = "KES"): string {
  return new Intl.NumberFormat("en-KE", {
    style: "currency",
    currency,
    minimumFractionDigits: 2,
  }).format(cents / 100);
}

/** "2,000" / "2000.50" -> cents. Throws on anything that isn't a clean amount. */
export function parseToCents(input: string): number {
  const cleaned = input.replace(/,/g, "").trim();
  const amount = Number(cleaned);
  if (!Number.isFinite(amount) || amount < 0) {
    throw new Error(`Not a valid amount: ${input}`);
  }
  return Math.round(amount * 100);
}

/**
 * Money for stat tiles: no minor units.
 *
 * A dashboard figure is read for magnitude, not reconciled to the cent, and
 * "Ksh 36,000.00" spends four characters saying nothing. The ledger keeps full
 * precision — this is only for headline numbers.
 */
export function formatCentsShort(cents: number, currency = "KES"): string {
  return new Intl.NumberFormat("en-KE", {
    style: "currency",
    currency,
    maximumFractionDigits: 0,
  }).format(cents / 100);
}

File: server/utils/money.py
from decimal import Decimal, InvalidOperation


def parse_amount_to_cents(amount_str: str) -> int:
    """'1,000.00' -> 100000. Raises ValueError on anything not a clean money string."""
    cleaned = amount_str.replace(",", "").strip()
    try:
        amount = Decimal(cleaned)
    except InvalidOperation as exc:
        raise ValueError(f"Not a valid amount: {amount_str!r}") from exc
    return int((amount * 100).to_integral_value())


def cents_to_display(cents: int) -> str:
    return f"{Decimal(cents) / 100:,.2f}"
